apps/prairielearn/src/pages/publicAssessments/publicAssessments.ts

```diff
@@ -27,6 +28,9 @@ router.get(
       throw new error.HttpStatusError(404, 'Course instance not public.');
     }
 
+    res.locals.user = res.locals.authn_user; // Need user to get copy targets. On public URLs, authn_user and user are always equal
+    const courseInstanceCopyTargets = await getCourseInstanceCopyTargets(res);
```

Comment:
Part of the reason that the `setQuestionCopyTarget` function just took an entire `Response` object was that it actually wrote something to `res.locals`. Now that values are just being returned, I'd much rather we have the functions accept a normal object of objects with whatever specific (typed) data it needs.